pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -38,29 +38,25 @@ func (c *Controller) HandlePacketIn(pktIn *ofctrl.PacketIn) error {
 		return errors.New("empty packetin for Antrea Policy")
 	}
 
-	matches := pktIn.GetMatches()
-	// Get custom reasons in this packet-in.
-	match := getMatchRegField(matches, openflow.CustomReasonField)
-	customReasons, err := getInfoInReg(match, openflow.CustomReasonField.GetRange().ToNXRange())
-	if err != nil {
-		return fmt.Errorf("received error while unloading customReason from reg: %v", err)
+	if len(pktIn.UserData) < 2 {
+		return nil
 	}
-
-	// Use reasons to choose operations.
-	var checkCustomReason = func(customReasonMark *binding.RegMark) bool {
-		return customReasons&customReasonMark.GetValue() == customReasonMark.GetValue()
+	packetInOperations := pktIn.UserData[1]
```

Comment:
Good one. Actually, I thought this before. Because there is a chance that multiple cases should be executed I'm not sure if using `switch` will be better. Using `switch` it will look like this:
```
        packetInOperations := pktIn.UserData[1]
	// Choose operations.
	var checkOperation = func(operation uint8) bool {
		return packetInOperations&operation == operation
	}
	switch {
	case checkOperation(openflow.PacketInNPLoggingOperation):
		if err := c.logPacket(pktIn); err != nil {
			return err
		}
		fallthrough
	case checkOperation(openflow.PacketInNPRejectOperation):
		if err := c.rejectRequest(pktIn); err != nil {
			return err
		}
		fallthrough
	case checkOperation(openflow.PacketInNPDenyOperation):
		if err := c.storeDenyConnection(pktIn); err != nil {
			return err
		}
	}
```
I'm not insisting to keep the current format. Lmk you know you want to change to use `switch`.

Reply:
I would prefer this format:

```
	packetInOperations := pktIn.UserData[1]
	switch packetInOperations {
	case openflow.PacketInNPLoggingOperation:
		if err := c.logPacket(pktIn); err != nil {
			return err
		}
	case openflow.PacketInNPRejectOperation:
		if err := c.rejectRequest(pktIn); err != nil {
			return err
		}
	case openflow.PacketInNPDenyOperation:
		if err := c.storeDenyConnection(pktIn); err != nil {
			return err
		}
	}
```

Reply:
Thanks for the explanation, then I am OK with the current implementation.

Reply:
My bad. The code I provided above couldn't work and I think yours neither. For example, if the `pktIn.UserData[1] == 0b101`, then we should execute `c.logPacket(pktIn)` and `c.storeDenyConnection(pktIn)`. The code blocks above don't work as we expect. Using a `switch` here doesn't make things easy.

Reply:
OK, then please keep the original solution